### ISSUE TITLE
Secret object type for deeper integration

### ISSUE THREAD

**Issue by mgreenegit** [PowerShell/SecretManagement#212]

**Comment by JustinGrote** [comment]
It's a reasonably convenient approach, I've always like how Microsoft.Extensions.Configuration basically takes a lot of complex inputs and boils them down to a key-value store.

Thoughts:
1. What if I, as a user, don't want to use secretmanagement to give you my credential? You'd need to instruct module authors to use parametersets probably which is an even more complicated scenario.
2. What if I don't want to use the default vault? How am I going to specify it? And how is your class going to know?

**Comment by mgreenegit** [comment]
@JustinGrote good point. could have an optional parameter for vault name. I only run into that when I store the same name in multiple vaults, but it happens.
